python-sdk/actionml/__init__.py

```diff
@@ -0,0 +1,634 @@
+        to "item".
+        """
+        return self.async_create(
+            event="$delete",
+            entity_type="item",
+            entity_id=iid,
+            event_time=event_time)
+
+    def delete_item(self, iid, event_time=None):
+        """Delete an item."""
+        return self.async_delete_item(iid, event_time).get_response()
+
+    def async_record_user_action_on_item(self, action, uid, iid, properties=None, event_time=None):
+        """
+        Create a user-to-item action.
+        Wrapper of async_create function, setting entity_type to "user" and
+        target_entity_type to "item".
+        """
+        if properties is None:
+            properties = {}
+        return self.async_create(
+            event=action,
+            entity_type="user",
+            entity_id=uid,
+            target_entity_type="item",
+            target_entity_id=iid,
+            properties=properties,
+            event_time=event_time)
+
+    def record_user_action_on_item(self, action, uid, iid, properties=None, event_time=None):
+        """Create a user-to-item action."""
+        if properties is None:
+            properties = {}
+        return self.async_record_user_action_on_item(
+            action, uid, iid, properties, event_time).get_response()
+
+
+class DatasetClient(BaseClient):
+    def __init__(self, url: str = "http://localhost:8080", threads=1, qsize=0, timeout=5):
+        self.path = "/datasets"
+        super(DatasetClient, self).__init__(url, threads, qsize, timeout)
+
+    def async_get(self, dataset_id: str):
+        """
+        Asynchronously get an dataset info from PIO Kappa Server.
+        :param dataset_id:
+        :returns: AsyncRequest object.
+        """
+        request = AsyncRequest("GET", self._add_segment(dataset_id))
+        request.set_response_handler(self._ok_response_handler)
+        self._connection.make_request(request)
+        return request
+
+    def get(self, dataset_id: str):
+        return self.async_get(dataset_id).get_response()
+
+    def async_create(self, dataset_id: str = None):
+        """
+        Asynchronously create dataset.
+        :param dataset_id: 
+        :return: 
+        """
+        data = {}
+        if dataset_id is not None:
+            data['id'] = dataset_id
+
+        request = AsyncRequest("POST", self.path, **data)
+        request.set_response_handler(self._create_response_handler)
+        self._connection.make_request(request)
+        return request
+
+    def create(self, dataset_id: str = None):
+        return self.async_create(dataset_id).get_response()
+
+    def async_delete(self, dataset_id: str):
+        request = AsyncRequest("DELETE", self._add_segment(dataset_id))
+        request.set_response_handler(self._delete_response_handler)
+        self._connection.make_request(request)
+        return request
+
+    def delete(self, dataset_id: str):
+        return self.async_delete(dataset_id).get_response()
+
```

Comment:
@laser13 I was wrong above, all `/dataset` endpoints are removed. we moved `/datasets/resource-id/events` to `/engines/resource-id/events` creating and deleting datasets is not supported in any way. We do it through creating and removing engines. 

We need some way to mirror the input json into HDFS files for later playback because once it is parsed and validated it is no longer an immutable event stream. This is best done in the REST-server by saving all raw json to some HDFS location then also sending it to the `engine.input` also. We should turn this on and off via `engine.json` params see: https://github.com/actionml/pio-kappa/blob/admin/pio-kappa-refactoring.md